### PR TITLE
[PTFE-141] ✨ Implement GHAS

### PR DESCRIPTION
**Pull request by gaspardmoindrot**

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: javascript
```
Comment (tcarmet) [inline .github/workflows/codeql.yaml]: Shouldn't it be typescript?
```suggestion
          languages: typescript
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: package-lock.json [hunk 2]
```diff
@@ -85,6 +85,7 @@
         "@fortawesome/free-regular-svg-icons": "^5.15.3",
         "@fortawesome/free-solid-svg-icons": "^5.15.3",
         "@fortawesome/react-fontawesome": "^0.1.14",
+        "@js-temporal/polyfill": "^0.4.3",
```
Comment (tcarmet) [inline package-lock.json]: Why are we adding a new dependency?

**Review by tcarmet — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 2 above)
Comment (gaspardmoindrot) [inline package-lock.json:88]:
```suggestion
```

File: .github/workflows/codeql.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    name: Static analysis with CodeQL
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: typescript
```
Comment (tcarmet) [inline .github/workflows/codeql.yaml]: My bad, rechecked the repo and as I see some .js file it should be both right?
```suggestion
          languages: typescript, javascript
```

**Review by tcarmet — CHANGES_REQUESTED** [review]

**Review by rachedbenmustapha — APPROVED** [review]

**Review by tcarmet — APPROVED** [review]

**Review by MonPote — APPROVED** [review]

**Pull request merged**